packages/components/src/internal/util/measurement.ts

```diff
@@ -267,6 +267,18 @@ export function getMultiAltUnitKeys(unitTypeStrs: string[]): string[] {
     return getAltUnitKeys(unitTypeStr);
 }
 
+export function convertUnitsForInput(amount: number, unit: string, displayUnit: string): number {
+    if (!amount || !displayUnit || !unit) {
+        return amount;
+    }
+    const currentUnit: MeasurementUnit = MEASUREMENT_UNITS[unit.toLowerCase()];
+    const targetUnit: MeasurementUnit = MEASUREMENT_UNITS[displayUnit.toLowerCase()];
+    if (!currentUnit || !targetUnit) {
+        return amount;
+    }
+    return parseFloat((amount * (currentUnit.ratio / targetUnit.ratio)).toFixed(6));
```

Comment:
any reason to have `convertUnitDisplay` call this function at line 307 to avoid duplicating this calculation? I can't quite decide if that would make sense or not given the other differences in these two functions.

Reply:
Yeah, I debated this too. Using the new method doesn't really simplify the logic for the other method at all, but at least the conversion is done in one place. So, sort of a coin toss.